x-pack/plugins/observability_solution/apm/server/routes/storage_explorer/has_storage_explorer_privileges.ts

```diff
@@ -18,12 +18,22 @@ export async function hasStorageExplorerPrivileges({
   apmEventClient: APMEventClient;
 }) {
   const {
-    indices: { transaction, span, metric, error },
+    // Only use apm index patterns and ignore OTel, as the storage explorer only supports APM data
+    indices: {
+      transaction = 'traces-apm*,apm-*',
+      span = 'traces-apm*,apm-*',
+      metric = 'metrics-apm*,apm-*',
+      error = 'logs-apm*,apm-*',
+    },
   } = apmEventClient;
 
   const names = uniq(
     [transaction, span, metric, error].flatMap((indexPatternString) =>
-      indexPatternString.split(',').map((indexPattern) => indexPattern.trim())
+      indexPatternString
+        .split(',')
+        .map((indexPattern) => indexPattern.trim())
+        // At this point we do not do any work for storage explorer + OTel data. So remove any otel related index
+        .filter((indexPattern) => !indexPattern.includes('otel'))
```

Comment:
I'm pretty sure this is too broad, eg it will exclude services that have otel in their service name (for whatever reason).

Reply:
Opened a bug actually. https://github.com/elastic/kibana/issues/193878